### PR TITLE
ViT + src.models.model refactor 

### PR DESCRIPTION
**Pull request by TheMikeMerrill**
This PR adds the ViT model *AND* refactors src.models.model from a script to a submodule. This shouldn't impact imports from anywhere else in the project, but will give us more flexibility as the project grows to include more models.

One thing to note is that it adds a new dependency - [einops](https://github.com/arogozhnikov/einops). At I don't like to add new packages if I can help it, but I think it's useful and hopefully others will too.

### CODE REVIEW

File: configs/models/ViTForClassification.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  accelerator: cuda
+  strategy: null
+  sync_batchnorm: false
+  precision: 32
+  enable_model_summary: true
+  weights_save_path: null
+  num_sanity_val_steps: 0
+  resume_from_checkpoint: null
+  profiler: null
+  benchmark: null
```
Comment (safranchik) [inline configs/models/ViTForClassification.yaml:43]: Setting `benchmark: true` can produce significant speedups so long as the input sizes don't vary. I think this is the case for most tasks in the repo?

File: configs/models/ViTForClassification.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  deterministic: null
```
Comment (safranchik) [inline configs/models/ViTForClassification.yaml:44]: Would setting `deterministic: true` as the default value make sense for reproducibility purposes?

**Review by safranchik — APPROVED** [review]

File: configs/models/ViTForClassification.yaml (hunk 1 above)
Comment (TheMikeMerrill) [reply configs/models/ViTForClassification.yaml:43]: Cool, I didn't know about this.  I'll consider making it a default for all models

File: configs/models/ViTForClassification.yaml (hunk 2 above)
Comment (TheMikeMerrill) [reply configs/models/ViTForClassification.yaml:44]: I guess deterministic:true is incompatible with benchmark:true
https://pytorch.org/docs/stable/notes/randomness.html

Not sure which way to go. If `benchmark:true` provides only modest performance improvements then let's use `deterministic: true`.

Comment (safranchik) [reply configs/models/ViTForClassification.yaml:44]: This is true. I would set `benchmark: true` as default, seeing how for most development runs determinism is unnecessary. It's only when reporting results that we want to set `deterministic: true`.

**Pull request merged**